File: Encapsulamiento y Constructores/02 - Taller 3 Python/Entregable/televisores/tv.py

class TV:
    _numTV = 0
    def __init__(self, marca, estado):
        self._marca = marca
        self._estado = estado
        self._canal = 1
        self._precio = 500
        self._volumen = 1
        self._control = None
        TV._numTV += 1

    
    def getMarca(self):
        return self._marca
    def setMarca(self, marca):
        self._marca = marca
    
    def getControl(self):
        return self._control
    def setControl(self, control):
        self._control = control
    
    def getPrecio(self):
        return self._precio
    def setPrecio(self, precio):
        self._precio = precio

    def getVolumen(self):
        return self._volumen
    def setVolumen(self, volumen):
        if self._volumen >= 0 and self._volumen <= 7 and self._estado == True:
            self._volumen = volumen
    
    def getCanal(self):
        return self._canal
    def setCanal(self, canal):
        if canal >= 1 and canal <= 120 and self._estado == True:
            self._canal = canal

    @classmethod
    def getNumTV(cls):
        return cls._numTV
    @classmethod
    def setNumTV(cls, numTV):
        cls._numTV = numTV
    
    def turnOn(self):
        self._estado = True
    def turnOff(self):
        self._estado = False
    
    def getEstado(self):
        return self._estado
    
    def canalUp(self):
        if self._canal < 120 and self._estado == True:
            self._canal += 1
    def canalDown(self):
        if self._canal > 1 and self._estado == True:
            self._canal -= 1
    def volumenUp(self):
        if self._volumen < 7 and self._estado == True:
            self._volumen += 1
    def volumenDown(self):
        if self._volumen > 0 and self._estado == True:
            self._volumen -= 1

File: Encapsulamiento y Constructores/02 - Taller 3 Java/Entregable/src/taller3/televisores/TV.java
package taller3.televisores;

public class TV {
	private Marca marca;
	private int canal = 1;
	private int precio = 500;
	private boolean estado;
	private int volumen = 1;
	private Control control;
	private static int numTV;
	
	public TV(Marca marca, boolean estado){
		this.marca = marca;
		this.estado = estado;
		TV.numTV += 1;
	}
	
	public Marca getMarca() {
		return marca;
	}
	public void setMarca(Marca marca) {
		this.marca = marca;
	}
	
	public Control getControl() {
		return control;
	}
	public void setControl(Control control) {
		this.control = control;
	}
	
	public int getPrecio() {
		return precio;
	}
	public void setPrecio(int precio) {
		this.precio = precio;
	}
	
	public int getVolumen() {
		return volumen;
	}
	public void setVolumen(int volumen) {
		if (volumen >= 0 && volumen <= 7 && estado == true) {
			this.volumen = volumen;
		}
	}
	
	public int getCanal() {
		return canal;
	}
	public void setCanal(int canal) {
		if (canal >= 1 && canal <= 120 && estado == true) {
			this.canal = canal;
		}
	}
	
	public static int getNumTV() {
		return numTV;
	}
	public static void setNumTV(int numTV) {
		TV.numTV = numTV;
	}
	
	public void turnOn() {
		estado = true;
	}
	public void turnOff() {
		estado = false;
	}
	
	public boolean getEstado() {
		return estado;
	}
	
	public void canalUp() {
		if (canal < 120 && estado == true) {
			canal += 1;
		}
	}
	public void canalDown() {
		if (canal > 1 && estado == true) {
			canal -= 1;
		}
	}
	
	public void volumenUp() {
		if (volumen < 7 && estado == true) {
			volumen += 1;
		}
	}
	public void volumenDown() {
		if (volumen > 0 && estado == true) {
			volumen -= 1;
		}
	}
}